src/tribler/core/components/database/db/layers/tests/test_user_activity_layer.py

```diff
@@ -0,0 +1,201 @@
+        assert float_equals(loser_2.preference, 0.0)
+        assert float_equals(loser_3.preference, 0.0)
+
+
+def test_store_weighted_decay(layer: UserActivityLayer) -> None:
+    """
+    Test result decay after updating.
+    """
+    layer.store("test query", InfoHash(b"\x00" * 20), {InfoHash(b"\x01" * 20),
+                                                       InfoHash(b"\x02" * 20),
+                                                       InfoHash(b"\x03" * 20)})
+    layer.store("test query", InfoHash(b"\x01" * 20), {InfoHash(b"\x00" * 20),
+                                                       InfoHash(b"\x02" * 20),
+                                                       InfoHash(b"\x03" * 20)})
+
+    with db_session():
+        queries = layer.Query.select()[:]
+        entry_1, = layer.InfohashPreference.select(lambda x: x.infohash == b"\x00" * 20)[:]
+        entry_2, = layer.InfohashPreference.select(lambda x: x.infohash == b"\x01" * 20)[:]
+        entry_3, = layer.InfohashPreference.select(lambda x: x.infohash == b"\x02" * 20)[:]
+        entry_4, = layer.InfohashPreference.select(lambda x: x.infohash == b"\x03" * 20)[:]
+
+        assert len(queries) == 1
+        assert queries[0].query == "test query"
+        assert float_equals(entry_1.preference, 0.2)
+        assert float_equals(entry_2.preference, 0.8)
+        assert float_equals(entry_3.preference, 0.0)
+        assert float_equals(entry_4.preference, 0.0)
+
+
+def test_store_delete_old(layer: UserActivityLayer) -> None:
+    """
+    Test result decay after updating.
+    """
+    layer.store("test query", InfoHash(b"\x00" * 20), {InfoHash(b"\x01" * 20),
+                                                       InfoHash(b"\x02" * 20),
+                                                       InfoHash(b"\x03" * 20)})
+    layer.store("test query", InfoHash(b"\x04" * 20), {InfoHash(b"\x00" * 20),
+                                                       InfoHash(b"\x01" * 20),
+                                                       InfoHash(b"\x02" * 20)})
+
+    with db_session():
+        queries = layer.Query.select()[:]
+        entry_1, = layer.InfohashPreference.select(lambda x: x.infohash == b"\x00" * 20)[:]
+        entry_2, = layer.InfohashPreference.select(lambda x: x.infohash == b"\x01" * 20)[:]
+        entry_3, = layer.InfohashPreference.select(lambda x: x.infohash == b"\x02" * 20)[:]
+        should_be_dropped = layer.InfohashPreference.select(lambda x: x.infohash == b"\x03" * 20)[:]
+        entry_4, = layer.InfohashPreference.select(lambda x: x.infohash == b"\x04" * 20)[:]
+
+        assert len(queries) == 1
+        assert queries[0].query == "test query"
+        assert float_equals(entry_1.preference, 0.2)
+        assert float_equals(entry_2.preference, 0.0)
+        assert float_equals(entry_3.preference, 0.0)
+        assert should_be_dropped == []
+        assert float_equals(entry_4.preference, 0.8)
```

Comment:
NIT: the version avoids unnecessary select querying and list copying:

```suggestion
        test_query = layer.Query.get()
        entry_1 = layer.InfohashPreference.get(lambda x: x.infohash == b"\x00" * 20)
        entry_2 = layer.InfohashPreference.get(lambda x: x.infohash == b"\x01" * 20)
        entry_3 = layer.InfohashPreference.get(lambda x: x.infohash == b"\x02" * 20)
        entry_4 = layer.InfohashPreference.get(lambda x: x.infohash == b"\x04" * 20)
        should_be_dropped = layer.InfohashPreference.get(lambda x: x.infohash == b"\x03" * 20)

    assert test_query.query == "test query"
    assert float_equals(entry_1.preference, 0.2)
    assert float_equals(entry_2.preference, 0.0)
    assert float_equals(entry_3.preference, 0.0)
    assert float_equals(entry_4.preference, 0.8)
    assert not should_be_dropped
```